src/components/Button/Button.tsx

```diff
@@ -3,9 +3,46 @@ import { IconAfter, IconBefore, Root } from './styles';
 import { ButtonProps } from './types';
 
 const Button: React.FC<ButtonProps> = (props) => {
-  const { iconBefore, children, iconAfter, loading, loadingText, as, target, ...rest } = props;
+  const {
+    iconBefore,
+    children,
+    iconAfter,
+    loading,
+    loadingText,
+    as,
+    target,
+    variant,
+    isDisabled,
+    btnPrimaryBgColor,
+    btnPrimaryTextColor,
+    btnSecondaryBgColor,
+    btnSecondaryTextColor,
+    btnConfirmedBgColor,
+    btnConfirmedTextColor,
```

Comment:
I think we don't need these props as we already have a way to customize colors via theme object. in theme ( src/theme/index.tsx ) we have specific configuration for Button component like this
```js
button: {
  primary: {
    background: philippineYellow,
    color: black,
  },
  secondary: {
    background: chineseBlack,
    color: white,
  },
  outline: {
    borderColor: philippineYellow,
    color: black,
  },
  plain: {
    color: black,
  },
  disable: {
    background: platinum,
    color: darkSilver,
  },
  confirmed: {
    background: oceanBlue,
    color: oceanBlue,
    borderColor: oceanBlue,
  },
},
```

So if someone wants to configure button color it will be handled at theme level as 99% projects will have kind of same button styles to every screens.

We want to use same kind of strategy to all components. Like CurrencyInput, Drawer, Loader etc. Please update all components using theme instead of props. 

then in future if we wanted to give granular customization then we will add `classname` or `style` props where user can pass fully customized styles.

please let me know if you have any doubt

Reply:
@SarjuHansaliya You're talking about all the reusable components which are located outside the swap widget, right?
So, only swap widget's main background and title text color will be editable via props here?

Reply:
for SwapWidget also we can add colors to theme object

```js
swapWidget: {
  // all necessary colors & fields 
}
```

Reply:
So, we have swap widget and buttons already set via theme object. It's the other components that are used within the swap widget are what need to be set in the similar way?

Reply:
We don't have swap widget colors via specific swap widget theme object. Right now we only have specific theme colors only for Button, ToggleButton and Switch.

![Screenshot 2022-08-22 at 5 22 33 PM](https://user-images.githubusercontent.com/6604146/185914984-f8ab65b5-76f6-4369-b896-d1eb4705d567.png)

Reply:
So what we want right now is swapWidget object with just main background and title text color. And separate individual objects for these (Button, CurrencyInput, Drawer, Loader, NumberOptions, Switch, TextInput, ToggleButtons), which don't exist already. Is this correct?

Reply:
yes correct @dev-catalyst . 

> And separate individual objects for these (Button, CurrencyInput, Drawer, Loader, NumberOptions, Switch, TextInput, ToggleButtons), which don't exist already.

One small correction, For Button, Switch, ToggleButton we already have object, but you can update it if you want.

Reply:
Closing this PR, new PR raised here: https://github.com/pangolindex/components/pull/100